paddle/memory/detail/CMakeLists.txt

```diff
@@ -0,0 +1,5 @@
+if(${WITH_GPU})
```

Comment:
nv_test was designed to handle *.cu files.  But here it is a .cc file, because it doesn't contain CUDA code.  However, here the source code calls cudaMallocHost defined in CUDA libraries, but we don't have a external/cuda.cmake.

I think a complete solution here should be a single line:

```cmake
cc_library(cpu_allocator_test SRCS cpu_allocator_test.cc DEPS cuda cudart)
```

Actually, I tried and succeeded to add `cmake/external/cuda.cmake` that defines two CMake targets -- cuda and cudart, but the building of `cpu_allocator_test` target complains that it cannot find `libpthread`.